cornice/tests/validationapp.py

```diff
@@ -187,20 +204,27 @@ def dummy_deserializer(request):
         values = request.body.decode().split(',')
         return dict(zip(['foo', 'bar', 'yeah'], values))
 
-    @custom_deserializer_service.post(schema=FooBarSchema,
-                                      deserializer=dummy_deserializer)
+    @custom_deserializer_service.post(validators=(schema_validation,))
     def custom_deserializer_service_post(request):
         return {"test": "succeeded"}
 
     class StringSequence(SequenceSchema):
         _ = SchemaNode(String())
 
     class ListQuerystringSequence(MappingSchema):
-        field = StringSequence(location="querystring")
+        field = StringSequence()
+
+        def deserialize(self, cstruct):
+            if 'field' in cstruct and not isinstance(cstruct['field'], list):
+                cstruct['field'] = [cstruct['field']]
+            return MappingSchema.deserialize(self, cstruct)
+
+    class QSSchema(MappingSchema):
+        query_string = ListQuerystringSequence()
 
-    @foobaz.get(schema=ListQuerystringSequence)
+    @foobaz.get(validators=(validator(QSSchema),))
     def foobaz_get(request):
-        return {"field": request.validated['field']}
+        return {"field": request.validated['query_string']['field']}
```

Comment:
I think it makes more sense but we have to document precisely where does this `validated` value comes from. First level keys are directly related to what the configured schema validators returns, right ?